src/trustRegion.jl

```diff
@@ -312,11 +324,12 @@ function perform_step!(cache::TrustRegionCache{false})
     cache.u_tmp = u .+ cache.step_size
     cache.fu_new = f(cache.u_tmp, p)
 
-    trust_region_step!(cache)
+    @unpack radius_update_scheme = cache
+    trust_region_step!(cache, Val(Int(radius_update_scheme)))
     return nothing
 end
 
-function trust_region_step!(cache::TrustRegionCache)
+function trust_region_step!(cache::TrustRegionCache, ::Val{0}) # conventional radius update scheme 
```

Comment:
Make a single dispatch and branch

Reply:
yes

Reply:
cool! also if I need some helper functions, like in Hei's scheme, there is a monotonic function used which depends on the actual:predicted change ratio. So these kind of helper functions can go in `utils.jl`, right?